README.md

```diff
@@ -1,21 +1,64 @@
 [![Build Status](https://ci.bazel.io/buildStatus/icon?job=BUILD_file_generator)](https://ci.bazel.io/job/BUILD_file_generator)
 
-# BUILD File Generator
+# BUILD File Generator (BFG)
 
-BUILD File Generator generates Bazel BUILD files for Java code.
+BUILD File Generator, or BFG for short, is a tool for generating Bazel BUILD files from one's source code. Instead of manually going through your project and creating BUILD files and their appropriate build rules, you can rely on BFG to do it for you! 
 
-1. It reads all `.java` files, and extracts the class dependency graph.
-2. Computes the strongly connected components of the graph.
-3. For each component, creates a `java_library` rule.
+An example project can be found here. TODO
+
+### Why is it useful?
+
+Bazel promises fast and correct builds, _especially_ for incremental builds. Rather than rebuild your entire codebase, Bazel _only_ rebuilds what is necessary, the targets that you have changed. For the remainder of your code, it relies on cached versions. 
```

Comment:
I'm super late to the party, so feel free to reject. 
We also have different writing styles (and I'm not a native English speaker anyway) so there's probably no "better" approach here.  

I suggest the following as a replacement for this section -

"""
Having all sources in a single BUILD rule doesn't allow Bazel to parallelize and cache builds. In order to fully benefit from Bazel, one must write multiple BUILD rules and connect them.

This project automates writing granular BUILD rules that allow Bazel to parallelize and cache builds. 

It's useful to quickly try out Bazel on your project, as well as to periodically optimize your build graph.
"""